types/src/on_chain_config.rs

```diff
@@ -7,10 +7,61 @@ use crate::{
     language_storage::{StructTag, TypeTag},
     transaction::SCRIPT_HASH_LENGTH,
 };
-use anyhow::Result;
+use anyhow::{format_err, Result};
 use libra_crypto::HashValue;
 use move_core_types::identifier::Identifier;
 use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use std::collections::HashMap;
+
+/// To register an on-chain config in Rust:
+/// 1. Implement OnChainConfig trait for the Rust representation of the config
+/// 2. Add an ID in `ConfigID` and update functions
+/// 3. Add getter method to `OnChainConfigPayload`
+
+#[derive(Copy, Clone, Eq, Hash, PartialEq)]
+pub enum ConfigID {
+    ScriptWhitelist,
+    // register more configs here
+}
+
+impl ConfigID {
+    pub fn all_configs() -> Vec<Self> {
+        vec![ConfigID::ScriptWhitelist]
+    }
+
+    pub fn name(self) -> String {
```

Comment:
this and above smell so bad

Reply:
at least for the above, the alternative would be to use this crate for iterating through static enums - otherwise I don't think there is a native Rust way to iterate through enums, even static ones 😢 https://stackoverflow.com/questions/21371534/in-rust-is-there-a-way-to-iterate-through-the-values-of-an-enum

Reply:
well, that's not introduced in this version of PR
previous one also had name function
it looks redundant and might be annoying to maintain
we can use some external crate that adds support automatically for iter and name
for example: https://substrate.dev/rustdocs/v1.0/strum/index.html